src/dynarec/la64/dynarec_la64_660f.c

```diff
@@ -1098,43 +1193,48 @@ uintptr_t dynarec64_660F(dynarec_la64_t* dyn, uintptr_t addr, uintptr_t ip, int
                     u8 = F8;
                     v0 = fpu_get_scratch(dyn);
                     v1 = fpu_get_scratch(dyn);
-                    VXOR_V(v0, v0, v0);
-                    VXOR_V(v1, v1, v1);
-                    if (u8 & 0x10) {
-                        VEXTRINS_W(v0, q0, 0x00);
-                        VEXTRINS_W(v1, q1, 0x00);
-                    }
-                    if (u8 & 0x20) {
-                        VEXTRINS_W(v0, q0, 0x11);
-                        VEXTRINS_W(v1, q1, 0x11);
-                    }
-                    if (u8 & 0x40) {
-                        VEXTRINS_W(v0, q0, 0x22);
-                        VEXTRINS_W(v1, q1, 0x22);
-                    }
-                    if (u8 & 0x80) {
-                        VEXTRINS_W(v0, q0, 0x33);
-                        VEXTRINS_W(v1, q1, 0x33);
-                    }
-                    VFMUL_S(v0, v0, v1);
-                    VPACKOD_W(v1, v0, v0);
-                    VPACKEV_W(v0, v0, v0);
-                    VFADD_S(v0, v0, v1);
-                    VPACKOD_D(v1, v0, v0);
-                    VPACKEV_D(v0, v0, v0);
-                    VFADD_S(v0, v0, v1);
-                    VXOR_V(q0, q0, q0);
-                    if (u8 & 0x1) {
-                        VEXTRINS_W(q0, v0, 0x00);
+                    v2 = fpu_get_scratch(dyn);
+                    VFMUL_S(v0, q0, q1);
+                    VXOR_V(v2, v2, v2);
+                    for (int i = 0; i < 4; ++i) {
+                        if (!(u8 & (1 << (4 + i)))) {
+                            VEXTRINS_W(v0, v2, (i << 4));
+                        }
                     }
-                    if (u8 & 0x2) {
-                        VEXTRINS_W(q0, v0, 0x11);
+                    VSHUF4I_W(v1, v0, 0b10110001); // v0[a,b,c,d] v1[b,a,d,c]
+                    VFADD_S(v0, v0, v1);           // v0[ab,ba,cd,dc]
+                    VSHUF4I_W(v1, v0, 0b01001110); // v1[cd,dc,ab,ba]
+                    VFADD_S(v0, v0, v1);           // v0[abcd,badc,cdab,dcba]
+                    VREPLVEI_W(q0, v0, 0);
```

Comment:
What's the purpose of `VREPLVEI_W`?

Reply:
different add order will cause  different  Nan/inf , +Nan or -Nan 
if use pickev/pickod  to strict  (a+b)+(c+d) , vreplvei_w is not needed.
if use vshuf4i.w a vreplvei.w/d is needed for dpps/dppd.
since pickev/pickod could be issue at same time, maybe it will be faster?